internal/k8s/probe.go

```
@@ -71,6 +73,40 @@ func (pm *ProbeManager) ValidateURL(targetURL string) error {
+func (pm *ProbeManager) CheckConnectivity(targetURL string, config BlackboxProbingConfig) bool {
+	if config.ProberURL == "" {
+		logger.Debugf("No prober URL configured, skipping pre-flight check")
+		return true
+	}
+
+	probeURL := fmt.Sprintf("http://%s/probe?target=%s&module=%s",
+		config.ProberURL,
+		url.QueryEscape(targetURL),
+		url.QueryEscape(config.Module))
+
+	resp, err := pm.httpClient.Get(probeURL)
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)